crates/core/Cargo.toml

```diff
@@ -19,6 +19,7 @@ reth-rlp = { workspace = true }
 reth-rpc-api = { workspace = true }
 reth-rpc-types = { workspace = true }
 starknet = { workspace = true }
+dojo-test-utils = { workspace = true }
```

Comment:
i think it makes sense to have this as a dev dependency, might save us some headache when we get to the point where we need to build releases

Reply:
more clarity around: my concern/question about having a dev-dependency:  

can it give us a clean boundary for test-compilation and release-compilation? 

from what ive been experiencing in rust you can have one entry into cargo work ok, but running cargo in a subcrate will not be able to compile, or developing for release will have a different feature flag which also breaks compilation